pyseir/icu/infer_icu.py

```diff
@@ -0,0 +1,107 @@
+import json
+import structlog
+import os.path
+
+import pandas as pd
+
+from pyseir import DATA_DIR
+import libs.datasets.combined_datasets as combined_datasets
+
+logger = structlog.get_logger()
+
+
+class ICUConfig:
+    WEIGHTS_PATHS = dict(
+        population=os.path.join(DATA_DIR, "population_weights_via_fips.json"),
+        one_month_trailing_cases=os.path.join(DATA_DIR, "one_month_trailing_weights_via_fips.json"),
+    )
+    LR_COEF = dict(m_hospitalized=0.2885, b=-1.6083)
+    SUFFIX = "_superset"
+    LOOKBACK_DAYS = 91
+    LOOKBACK_DATE = pd.Timestamp.today() - pd.Timedelta(days=LOOKBACK_DAYS)
+
+
+def get_data_for_icu_calc(fips: str) -> pd.DataFrame:
+    """
+    Get the timeseries data for the current aggregation level and the superset aggregation level.
+    In the case where the current aggregation level is the highest (state), return the superset as
+    state too.
+    """
+    COLUMNS = ["cases", "deaths", "current_icu", "current_hospitalized"]
+    f = combined_datasets.get_timeseries_for_fips
+    this_level_df = (
+        f(fips, columns=COLUMNS).get_subset(after=ICUConfig.LOOKBACK_DATE).data.set_index("date")
+    )
+    super_level_df = (
+        f(fips[:2], columns=COLUMNS)
+        .get_subset(after=ICUConfig.LOOKBACK_DATE)
+        .data.set_index("date")
+    )
+    return pd.merge(
```

Comment:
I found the date alignment helpful in debugging, since county and state reported values rarely align perfectly. I'd prefer to leave this for now, and readdress once we decide what to do in web adaptor.